app/services/openlibrary_service.rb

```diff
@@ -0,0 +1,34 @@
+require 'set'
+
+class OpenlibraryService
```

Comment:
Functionally, this service it's ok. But `search_by_isbn(isbn)` and `parse_response_book(book, isbn)` are too long methods. 
We should delegate if to many methods [check long method refactoring](https://sourcemaking.com/refactoring/smells/long-method)